src/GitExtensions.PluginTemplate/Plugin.cs

```diff
@@ -1,34 +1,48 @@
 ﻿using GitExtensions.PluginTemplate.Properties;
 using GitUIPluginInterfaces;
 using ResourceManager;
-using System;
-using System.Collections.Generic;
 using System.ComponentModel.Composition;
-using System.Diagnostics;
-using System.IO;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using System.Windows.Forms;
 
 namespace GitExtensions.PluginTemplate
 {
     /// <summary>
-    /// An empty Git Extensions plugin implementation.
+    /// A template for Git Extensions plugins.
     /// </summary>
     [Export(typeof(IGitPlugin))]
     public class Plugin : GitPluginBase
     {
         public Plugin()
         {
-            Name = "PluginTemplate";
-            Description = "Plugin Template";
+            SetNameAndDescription("Plugin Template");
             Icon = Resources.Icon;
         }
 
-        public override bool Execute(GitUIEventArgs e)
+        
+        /// <summary>
+        /// This method is called when the plugin is loaded (either when Git Extensions is opened or when repositories are switched).
+        /// </summary>
+        public override void Register(IGitUICommands gitUiCommands)
         {
-            MessageBox.Show(e.OwnerForm, "Hello from the Plugin Template.", "Git Extensions");
+            

```

Comment:
May be a good idea to put a link to our wiki page where we can explain the method and give some ideas what this method is good for.

Reply:
Good idea, I will add this.

Reply:
I drafted a wiki page and will add links to its sections:
https://github.com/gitextensions/gitextensions.plugintemplate/wiki/GitPluginBase

Reply:
done